src/main/java/com/exactpro/th2/act/ActHandler.java

```diff
@@ -169,6 +177,47 @@ public void placeOrderCancelReplaceRequest(PlaceMessageRequest request, StreamOb
         }
     }
 
+    @Override
+    public void sendRawMessage(SendRawMessageRequest request, StreamObserver<SendRawMessageResponse> responseObserver) {
+        long startPlaceMessage = System.currentTimeMillis();
+        try {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Sending  message request: " + shortDebugString(request));
+            }
+
+            String actName = "sendRawMessage";
+            // FIXME store parent with fail in case of children fail
+            EventID parentId = createAndStoreParentEvent(request, actName, PASSED);
+
+            Checkpoint checkpoint = registerCheckPoint(parentId);
+
+            if (Context.current().isCancelled()) {
+                LOGGER.warn("'{}' request cancelled by client", actName);
+                sendRawMessageErrorResponse(responseObserver, "Request has been cancelled by client");
+            }
+
+            try {
+                sendRawMessage(request.getRaw(), parentId);
+            } catch (Exception ex) {
+                createAndStoreErrorEvent("sendMessage", ex.getMessage(), Instant.now(), parentId);
+                throw ex;
+            }
+
+            SendRawMessageResponse response = SendRawMessageResponse.newBuilder()
+                .setStatus(RequestStatus.newBuilder().setStatus(SUCCESS).build())
+                .setCheckpointId(checkpoint)
+                .build();
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+
+        } catch (RuntimeException | IOException e) {
```

Comment:
I think we should catch `Exception` here. Otherwise, we might no complete the call and it will be stuck until the application is down